tests/unit/test_approximation.py

```diff
@@ -74,555 +79,228 @@ def setUp(self) -> None:
                 0.4981814349432025,
                 0.3670700196710188,
             ]
         )
-
-    def test_kernel_mean_approximator_creation(self) -> None:
-        """
-        Test the class KernelMeanApproximator initialises correctly.
-        """
-        # Disable pylint warning for abstract-class-instantiated as we are intentionally
-        # patching these whilst testing creation of the parent class
-        # pylint: disable=abstract-class-instantiated
-        # Patch the abstract method (approximate) of the KernelMeanApproximator, so it
-        # can be created
-        p = patch.multiple(
-            coreax.approximation.KernelMeanApproximator, __abstractmethods__=set()
-        )
-        p.start()
-
-        # Define the approximator
-        approximator = coreax.approximation.KernelMeanApproximator(
-            kernel=self.kernel,
-            random_key=self.random_key,
-            num_kernel_points=self.num_kernel_points,
+        return (
+            random_key,
+            data,
+            kernel,
+            num_kernel_points,
+            num_train_points,
+            true_distances,
         )
-        # pylint: enable=abstract-class-instantiated
-
-        # Check parameters have been set
-        self.assertEqual(approximator.kernel, self.kernel)
-        np.testing.assert_array_equal(approximator.random_key, self.random_key)
-        self.assertEqual(approximator.num_kernel_points, self.num_kernel_points)
 
-    def test_random_approximator(self) -> None:
+    def test_approximation_accuracy(
+        self, problem, approximator: type[coreax.approximation.KernelMeanApproximator]
+    ) -> None:
         """
-        Verify random approximator performance on toy problem.
+        Verify approximator performance on toy problem.
 
         This test verifies that, when the entire dataset is used to train the
         approximation, the result is very close to the true value. We further check if a
         subset of the data is used for training, that the approximation is still close
         to the true value, but less so than when using a larger training set.
         """
+        (
+            random_key,
+            data,
+            kernel,
+            num_kernel_points,
+            num_train_points,
+            true_distances,
+        ) = problem
         # Define the approximator - full dataset used to fit the approximation
-        approximator_full = coreax.approximation.RandomApproximator(
-            kernel=self.kernel,
-            random_key=self.random_key,
-            num_kernel_points=self.data.shape[0],
-            num_train_points=self.data.shape[0],
-        )
+        if not issubclass(approximator, coreax.approximation.NystromApproximator):
+            full_kwargs = {
+                "num_kernel_points": data.shape[0],
+                "num_train_points": data.shape[0],
+            }
+        else:
+            full_kwargs = {"num_kernel_points": data.shape[0]}
```

Comment:
My personal preference would be for these if-else blocks to be `if issubclass(...): ... else: ...`; that reads slightly clearer to me.

Reply:
(Same comment applies to the rest of the `if not issubclass` checks in this file!)

Reply:
I agree and have updated accordingly

Reply:
Looks good to me!